diff --git a/modules/openid/openid.module b/modules/openid/openid.module
index 498e935..0adc54b 100644
--- a/modules/openid/openid.module
+++ b/modules/openid/openid.module
@@ -103,7 +103,7 @@ function openid_user_insert(&$edit, $account, $category) {
 function openid_user_login(&$edit, $account) {
   if (isset($_SESSION['openid'])) {
     // The user has logged in via OpenID.
-    if(is_array($_SESSION['openid']['user_login_values'])) {
+    if(isset($_SESSION['openid']['user_login_values']) && is_array($_SESSION['openid']['user_login_values'])) {
       user_cookie_save(array_intersect_key($_SESSION['openid']['user_login_values'], array_flip(array('openid_identifier'))));
     }
     unset($_SESSION['openid']);
