diff --git a/sites/all/modules/contrib/views/handlers/views_handler_filter_combine.inc b/sites/all/modules/contrib/views/handlers/views_handler_filter_combine.inc
index c1afcf5..8d620b3 100644
--- a/sites/all/modules/contrib/views/handlers/views_handler_filter_combine.inc
+++ b/sites/all/modules/contrib/views/handlers/views_handler_filter_combine.inc
@@ -119,7 +119,7 @@ class views_handler_filter_combine extends views_handler_filter_string {
       $words = $phrase ? array($words) : preg_split('/ /', $words, -1, PREG_SPLIT_NO_EMPTY);
       $placeholder = $this->placeholder();
       foreach ($words as $word) {
-        $where->where($field . " LIKE $placeholder", array($placeholder => '%' . db_like(trim($word, " ,!?")) . '%'));
+		$where->where("LOWER({$field}) LIKE LOWER({$placeholder})", array($placeholder => '%' . db_like(trim($word, " ,!?")) . '%'));
       }
     }