--- modules/geolocation_googlemaps/geolocation_googlemaps_widget.js	2015-08-12 15:15:37.000000000 +0300
+++ modules/geolocation_googlemaps/geolocation_googlemaps_widget.js	2015-08-12 15:16:00.000000000 +0300
@@ -72,9 +72,12 @@
         Drupal.geolocation.setMapMarker(results[0].geometry.location, i);
         Drupal.geolocation.codeLatLng(results[0].geometry.location, i, 'textinput');
         Drupal.geolocation.setZoom(i, results[0].geometry.location_type);
+        $('.map-location-not-found').remove();
       }
       else {
-        alert(Drupal.t('Geocode was not successful for the following reason: ') + status);
+        $('.map-location-not-found').remove();
+        $('.field-name-field-schedule-city-id:visible').parent().find('.geolocation-address').append('<div class="map-location-not-found">Asukohta ei leitud järgneval põhjusel: ' + status + '</div>');
+        //alert(Drupal.t('Geocode was not successful for the following reason: ') + status);
       }
     });
   }
