diff --git a/sites/all/modules/contrib/geolocation/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js b/sites/all/modules/contrib/geolocation/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js
index b87d7e1..e9a2cad 100644
--- a/sites/all/modules/contrib/geolocation/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js
+++ b/sites/all/modules/contrib/geolocation/modules/geolocation_googlemaps/geolocation_googlemaps_widget.js
@@ -67,6 +67,7 @@
     }
 
     geocoder.geocode( { 'address': address }, function(results, status) {
+     if (address != "") {
       if (status == google.maps.GeocoderStatus.OK) {
         Drupal.geolocation.maps[i].setCenter(results[0].geometry.location);
         Drupal.geolocation.setMapMarker(results[0].geometry.location, i);
@@ -79,6 +80,7 @@
         $('.field-name-field-schedule-city-id:visible').parent().find('.geolocation-address').append('<div class="map-location-not-found">Asukohta ei leitud järgneval põhjusel: ' + status + '</div>');
         //alert(Drupal.t('Geocode was not successful for the following reason: ') + status);
       }
+     }
     });
   }
 
